extensions/amp-form/0.1/amp-form.js

```diff
@@ -16,14 +16,82 @@
 
 import {isExperimentOn} from '../../../src/experiments';
 import {getService} from '../../../src/service';
+import {assertHttpsUrl} from '../../../src/url';
+import {user} from '../../../src/log';
+import {onDocumentReady} from '../../../src/document-ready';
+import {xhrFor} from '../../../src/xhr';
+import {toArray} from '../../../src/types';
+import {startsWith} from '../../../src/string';
+
 
 /** @type {string} */
 const TAG = 'amp-form';
 
+class AmpForm {
+
+  /**
+   * Adds functionality to the passed form element and listens to submit event.
+   * @param element {!HTMLFormElement}
+   */
+  constructor(element) {
+    /** @const @private {!Window} */
+    this.win_ = element.ownerDocument.defaultView;
+
+    /** @const @private {!Element} */
+    this.form_ = element;
+
+    /** @const @private {!Xhr} */
+    this.xhr_ = xhrFor(this.win_);
+
+    /** @const @private {string} */
+    this.method_ = this.form_.getAttribute('method') || 'GET';
+
+    /** @const @private {string} */
+    this.xhrAction_ = this.form_.getAttribute('data-xhr-action');
+    if (this.xhrAction_) {
+      assertHttpsUrl(this.xhrAction_, this.form_, 'data-xhr-action');
+      user.assert(!startsWith(this.xhrAction_, 'https://cdn.ampproject.org'),
+          'form data-xhr-action should not be on cdn.ampproject.org: %s',
+          this.form_);
+    }
+    this.installSubmitHandler_();
+  }
+
+  /** @private */
+  installSubmitHandler_() {
+    this.form_.addEventListener('submit', e => {
+      if (e.defaultPrevented) {
+        return;
+      }
+      if (this.xhrAction_) {
+        this.xhr_.fetchJson(this.xhrAction_, {
+          body: new FormData(this.form_),
+          method: this.method_,
+        });
+        e.preventDefault();
+      }
+    });
+  }
+}
+
+
+/**
+ * Installs submission handler on all forms in the document.
+ * @param win {!Window}
+ */
+function installSubmissionHandlers(win) {
+  onDocumentReady(win.document, () => {
```

Comment:
I am cool either way and I think we can get it to work in either pattern, although I think the pattern that I have in this PR is the most readable and easier to add functionality that is encapsulated to a single form. 

Is the motivation behind not using this pattern is to just have a single submit handler that can handle all forms submit events? Just curious what are other motivations in doing it that way instead of what I am currently doing?